crates/stages/src/stages/execution.rs

```diff
@@ -86,19 +87,24 @@ impl<EF: ExecutorFactory> ExecutionStage<EF> {
     // TODO: This should be in the block provider trait once we consolidate
     // SharedDatabase/Transaction
     fn read_block_with_senders<DB: Database>(
-        tx: &Transaction<'_, DB>,
+        provider: &reth_provider::DatabaseProviderRW<'_, DB>,
         block_number: BlockNumber,
     ) -> Result<(BlockWithSenders, U256), StageError> {
-        let header = tx.get_header(block_number)?;
-        let td = tx.get_td(block_number)?;
-        let ommers = tx.get::<tables::BlockOmmers>(block_number)?.unwrap_or_default().ommers;
-        let withdrawals = tx.get::<tables::BlockWithdrawals>(block_number)?.map(|v| v.withdrawals);
+        let header = provider
+            .header_by_number(block_number)?
+            .ok_or_else(|| ProviderError::HeaderNotFound(block_number.into()))?;
+        let td = provider
+            .header_td_by_number(block_number)?
+            .ok_or_else(|| ProviderError::HeaderNotFound(block_number.into()))?;
+        let ommers = provider.ommers(block_number.into())?.unwrap_or_default();
+        let withdrawals = provider.withdrawals_by_block(block_number.into(), header.timestamp)?;
```

Comment:
Is this header.timestamp required? It wasn't before, I guess it's to check if we're past the transition to short-circuit it? Do we do the same for querying header_td?

Reply:
> I guess it's to check if we're past the transition to short-circuit it?

yeah correct, there isn't for `header_td` but there is a TODO in the code for it as well